src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -30,7 +30,11 @@ private WellKnownTypeProvider(Compilation compilation)
             GenericTask = GetTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericTask);
             CollectionTypes = GetWellKnownCollectionTypes();
             SerializationInfo = GetTypeByMetadataName(WellKnownTypeNames.SystemRuntimeSerializationSerializationInfo);
-            GenericIEquatable = GetTypeByMetadataName(WellKnownTypeNames.SystemIEquatable1);
+            Array = GetTypeBySpecialType(SpecialType.System_Array);
```

Comment:
I have decided to add the special types as property too. If you prefer them to not be properties here I can create another commit to change it.